tenets/codelingo/go/writes-to-async-pointer/example.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+)
+
+type thing struct {
+	name string
+}
+
+func getNameFromPointer(t *thing) {
+	fmt.Println(t.name)
+}
+
+func getName(t thing) {
+	fmt.Println(t.name)
+}
+
+func setNameOnPointer(t *thing) {
+	t.name = "car"
+}
+
+func newThingPointer(name string) *thing {
+	return &thing{name}
+}
+
+func newThing(name string) thing {
+	return thing{name}
+}
+
+func main() {
+
+	t := newThingPointer("Pointer")
+
+	getNameFromPointer(t)
+
+	go getNameFromPointer(t)
```

Comment:
this block really makes me think we need a sense of read-sections and write-sections

would we catch `go read(); write()`? 'cos it's fundamentally the same issue

Reply:
Are you referring to the Read and Write methods of the io package?

Reply:
I'm referring to the operations in the test file which conflict if run at the same time, which I'd been calling `readVulnerableField`/`writeVulnerableField` above